src/object/srv_ec_aggregate.c

```diff
@@ -2212,6 +2212,48 @@ agg_obj_is_leader(struct ds_pool *pool, struct daos_oclass_attr *oca,
 	return rc;
 }
 
+static int
+agg_filter(daos_handle_t ih, vos_iter_desc_t *desc, void *cb_arg, unsigned int *acts)
+{
+	struct ec_agg_param	*agg_param = (struct ec_agg_param *)cb_arg;
+	struct ec_agg_entry	*agg_entry = &agg_param->ap_agg_entry;
+	struct ec_agg_pool_info	*info = &agg_param->ap_pool_info;
+	struct daos_oclass_attr  oca;
+	int			 rc = 0;
+
+	if (desc->id_type != VOS_ITER_OBJ)
+		return 0;
+
+	rc = dsc_obj_id2oc_attr(desc->id_oid.id_pub, &info->api_props, &oca);
+	if (rc) {
+		D_ERROR("Skip object("DF_OID") with unknown class(%u)\n",
+			DP_OID(desc->id_oid.id_pub),
+			daos_obj_id2class(desc->id_oid.id_pub));
+		*acts = VOS_ITER_CB_SKIP;
+		agg_param->ap_credits++;
+		goto done;
+	}
+
+	if (!daos_oclass_is_ec(&oca)) { /* Skip non-EC object */
+		D_DEBUG(DB_EPC, "Skip oid:"DF_UOID" non-ec obj\n",
+			DP_UOID(desc->id_oid));
+		agg_param->ap_credits++;
```

Comment:
it is a bit strange to only increase the credit for non-ec case here. Probably the credit check should be moved from agg_iterate_pre_cb to here completely.

Reply:
I thought part of the point was to account for time spent iterating the non-ec entries too.   Though perhaps we could add some priority such as one of those is worth 1 credit and actual EC object is worth 10 credits or some such.

Reply:
Using another counter "skipped_credits" could be easier for debugging?

Reply:
Yeah, good point.  This patch is getting rather large.  Can we defer that to another patch?